java/arcs/core/storage/StorageProxy.kt

```diff
@@ -254,6 +275,7 @@ class StorageProxy<Data : CrdtData, Op : CrdtOperationAtTime, T>(
      * Applies messages from a [Store].
      */
     suspend fun onMessage(message: ProxyMessage<Data, Op, T>) {
+        checkNotClosed()
```

Comment:
Is there a window of time here where the proxy is closed, but a pending message arrives from the ServiceStore?

Reply:
You're right, I think this will be an issue with the `ServiceStore`, for example, since the `off` call dispatches to a queue and doesn't wait.

Reply:
Another potential option: wire up `idle` for `ServiceStore`, and have the `StorageProxy` wait for the `off` operation to propagate.

Reply:
Another option would be to check if it is closed, and return (no-op) in this case, so just ignore incoming messages after close.   Although this might let leaks creep in that don't get detected where the ServiceStore keeps sending for a long time and the connection to it isn't ever shutdown?

Reply:
Oh strange, I swear in my first response I also typed "I will downgrade this to an informational log message", but those words are clearly not there.